src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -111,11 +111,13 @@ public boolean oauth2VerifyNonce(final String nonce) {
      *
      * @return The Nonce String from within the session or null if not found.
      */
-    private String getSessionNonce() {
+    private String getSessionNonce(Session chSession) {
         String oauth2Nonce = null;
         try {
-            final Map<String, Object> data = sessionService.getSessionDataFromContext();
+            final Map<String, Object> data = chSession.getData();
             oauth2Nonce = (String) data.getOrDefault(SessionKeys.NONCE.getKey(), null);
+            data.put(SessionKeys.NONCE.getKey(), "");
+            LOGGER.debug("Verify nonce against expected nonce from session");
```

Comment:
I don't understand this log message in this context.

Reply:
The debug message was taken from the Perl code example in https://github.com/companieshouse/CH-MojoX-SignIn/blob/develop/lib/CH/MojoX/SignIn/Plugin/OAuth2.pm#L46. If you have any recommendations for a better debug message that would be great.

Reply:
I'd suggest that it isn't verifying in this method, so if you do need a debug message there, then maybe "Obtained nonce from session".